File: common/src/test/java/dev/number6/generate/PresentableKeyPhrasesResultsGenerator.java
package dev.number6.generate;


import dev.number6.comprehend.results.PresentableKeyPhrasesResults;
import uk.org.fyodor.generators.Generator;

import java.time.LocalDate;
import java.util.Map;

public class PresentableKeyPhrasesResultsGenerator implements Generator<PresentableKeyPhrasesResults> {

    Generator<Map<String, Long>> keyPhrasesResultsGenerator = CommonRDG.map(CommonRDG.string(10),
            CommonRDG.longVal(100));

    @Override
    public PresentableKeyPhrasesResults next() {
        return new PresentableKeyPhrasesResults(LocalDate.now(), keyPhrasesResultsGenerator.next(), CommonRDG.string().next());
    }
}

File: common/src/test/kotlin/dev/number6/generate/PresentableKeyPhrasesResultsGenerator.kt
package dev.number6.generate

import dev.number6.comprehend.results.PresentableKeyPhrasesResults
import uk.org.fyodor.generators.Generator
import uk.org.fyodor.generators.RDG
import java.time.LocalDate

class PresentableKeyPhrasesResultsGenerator : Generator<PresentableKeyPhrasesResults> {
    var keyPhrasesResultsGenerator = RDG.map(RDG.string(10),
            RDG.longVal(100))

    override fun next(): PresentableKeyPhrasesResults {
        return PresentableKeyPhrasesResults(LocalDate.now(), keyPhrasesResultsGenerator.next(), RDG.string().next())
    }
}